### DISCUSSION TITLE
How should measure data and event_data interact

### DISCUSSION THREAD

**Discussion by ThomUK** [ThomUK/SPCreporter#84] [General]
Should it be possible to supply both wide measure_data, and long event_data for the same metric?

Probably not (otherwise it would be possible to accidentally supply completely different datasets for the same metric ref).  The wide aggregated data should be calculated from the event data.  This is the only way to guarantee that there is consistency over the data presented on the report.

Similarly, should measure_data contain both formats on the same worksheet / list?  Or should there be separate arguments for wide measure data and long event data like:
```
make_data_bundle(
  measure_data = wide_pre_aggregated_data, 
  event_data = long_event_level_data,
  ...
)
```

**Comment by francisbarton** [comment]
I'd be in favour of submitting wide aggregated data and long event data as separate arguments in the way you suggest, rather than as list.

I also don't think SPCr should insist on either of them existing - it should be able to cope with one of them being absent, as users will not necessarily always want to create a report with both kinds of data included.
I don't have a problem with supply wide and long data for the same metric. You might want to chart a metric in different ways over different time periods, hypothetically. I don't see the value in making SPCr override the user-supplied data by calculating the wide data from the long.
